Pony/source/Socket.cpp

```diff
@@ -79,58 +84,56 @@ bool Socket::Open(unsigned short port)
 
 #if defined(_WIN32)
     DWORD nonBlocking = 1;
-    if (ioctlsocket( socket, FIONBIO, &nonBlocking) != 0) {
-	printf( "failed to set non-blocking socket\n" );
-	Close();
-	return false;
-    }
+    if (ioctlsocket(m_socket, FIONBIO, &nonBlocking) != 0) {
 #else
     int nonBlocking = 1;
     if (fcntl(m_socket, F_SETFL, O_NONBLOCK, nonBlocking) == -1) {
+#endif
         Close();
         return false;
     }
-#endif
+
     return true;
 }
 
-int Socket::Send(const Address & dst, const void * data, unsigned size)
+signed Socket::Send(const Address& dst, const void* data, unsigned size)
 {
-    if ((m_socket == 0) || ( ! data) || (size <=0))
+    if ((m_socket == 0) || (!data) || (size <= 0))
```

Comment:
Size is unsigned. It cannot be less than zero.

Reply:
ok